test/integration/ChargedParticles.test.js

```diff
@@ -1500,4 +1511,120 @@ describe("[INTEGRATION] Charged Particles", () => {
+        contractCaller: signer1,
+        contractParams: [
+          user1,                        // creator
+          user2,                        // receiver
+          user1,                        // referrer
+          TEST_NFT_TOKEN_URI,           // tokenMetaUri
+          'reward',                     // walletManagerId
+          daiAddress,                   // assetToken
+          toWei('10'),                  // assetAmount
+          0,                            // annuityPercent
+        ],
+      });
+    
+      await tokenInfoProxyMock.mock.getTokenOwner.withArgs(proton.address, energizedParticleId.toString()).returns(user2);
+    
+      const energizedNftWalletManagerAddress = await rewardWalletManager.callStatic.getWalletAddressById(
+        proton.address, 
+        energizedParticleId,
+        proton.address, 
+        100
+      );
+      const initiatedStakeOnEnergized = await rewardProgram.walletStake(energizedNftWalletManagerAddress);
+      expect(initiatedStakeOnEnergized).to.have.property('start');
+    
+      // fund reward program.
+      await ionx.connect(protocolOwnerSigner).transfer(deployer, fundingAmount).then(tx => tx.wait());
+      await ionx.connect(signerD).approve(rewardProgram.address, fundingAmount).then(tx => tx.wait());
+      await rewardProgram.connect(signerD).fund(fundingAmount).then(tx => tx.wait());
+      const rewardProgramIonxBalance = await ionx.balanceOf(rewardProgram.address);
+
+      console.log(ethers.utils.formatUnits(rewardProgramIonxBalance, 18));
+
+      expect(rewardProgramIonxBalance).to.equal(fundingAmount);
+    
+      // Reverts not wallet manager
+      await expect(
+        rewardProgram.connect(signerD).unstake(
+          user2,
+          proton.address,
+          1,
+      )).to.be.revertedWith("Not wallet");
+
+      expect(await ionx.balanceOf(user2)).to.be.eq(0);
+
+      await setNetworkAfterBlockNumber(Number((await getNetworkBlockNumber()).toString()) + 2);
+
+      await chargedParticles.connect(signer2).releaseParticle(
+        user2,
+        proton.address,
+        energizedParticleId,
+        'reward',
+        daiAddress
+      );
+      expect(await dai.balanceOf(user2)).to.be.above(toWei('9.9'));
+      const stakeOnRelease = await rewardProgram.walletStake(energizedNftWalletManagerAddress);
+      expect(stakeOnRelease['generatedCharge']).gt(0);
+
+      console.log(ethers.utils.formatUnits(stakeOnRelease['generatedCharge'], 6), ethers.utils.formatUnits(stakeOnRelease['reward'], 18));
+    
+      expect(await ionx.balanceOf(user2)).to.be.eq(stakeOnRelease['reward']);
+    });
+
+    it ('Deployed wallet manager set in reward program', async function() {
+      const rewardWalletManager = getDeployData('RewardWalletManager', chainId);
+      expect(await rewardProgram.rewardWalletManager()).to.be.eq(rewardWalletManager.address);
+    });
+
+    it ('It records a second nft into the reward program', async function() {
```

Comment:
This last test when run produces this error.
<img width="842" alt="image" src="https://user-images.githubusercontent.com/40045101/224425684-2d4030a5-7ca0-41c7-bd4f-45f2c301f97d.png">